### ISSUE TITLE
mesa-demos is packaged "twice"

### ISSUE THREAD

**Issue by primeos** [NixOS/nixpkgs#89109]
We have two Nixpkgs packages for [mesa-demos](https://gitlab.freedesktop.org/mesa/demos):
- [pkgs/tools/graphics/mesa-demos/default.nix](https://github.com/NixOS/nixpkgs/blob/6b227dd56faaf5466a023e827149fb2018b15f07/pkgs/tools/graphics/mesa-demos/default.nix) (cc @andersk)
- [pkgs/tools/graphics/glxinfo/default.nix](https://github.com/NixOS/nixpkgs/blob/6b227dd56faaf5466a023e827149fb2018b15f07/pkgs/tools/graphics/glxinfo/default.nix) (cc @abbradar)

This might be intentionally as `glxgears` only packages a subset of the demos and has much fewer dependencies. But I think we should at least document this and ideally also at least de-duplicate the `src` attribute (so that the versions always match).

**Comment by matthewbauer** [comment]
Yeah - glxinfo is only glxinfo, while mesa-demos has everything in mesa-demos. It might be okay to just remove glxinfo and replace it with mesa-demos now though. Unsure how much bigger mesa-demos is compared to glxinfo.

**Comment by prusnak** [comment]
glxinfo ~ 160 KB
mesa-demos ~ 11 MB

**Comment by davidak** [comment]
it might make sense to have glxinfo as extra package. it has a different use than the demos

but one could also argue that we should package software as upstream intends it. it is actually not packaged this way often. only AUR, MacPorts and us https://repology.org/project/glxinfo/versions

**Comment by delroth** [comment]
The current glxinfo derivation is pretty hacky, it bypasses the build system entirely and hardcodes compilation flags and dependencies. +1 to getting rid of it. Will send a PR making it an alias.

10MB extra on systems that already carry enough stuff to need OpenGL seems like a fairly minimal cost. If someone cares I suggest they send a followup PR making glxinfo a derivation which properly subsets the files from the mesa-demos outputs.

**Comment by Atemu** [comment]
I think it'd be a good idea to make glxinfo and glxgears separate outputs of mesa-demos instead and expose them as top-level attrs via aliases.

That way you still get to choose to only install the minimal set most people would use to verify OGL support.
